meerk40t/core/cutplan.py

```diff
@@ -175,6 +175,10 @@ def preprocess(self):
         #     axis = rotary.axis
 
         original_ops = copy(self.plan)
+        if self.context.opt_raster_optimisation:
+            self.optimize_rasters(original_ops, "op raster")
+            # We could do this as well, but images are burnt separately anyway...
```

Comment:
There is actually a second potential optimisation algorithm which could be applied to large images which is a whitespace optimisation to chop images with large areas of whitespace (think of a large picture frame) into separate smaller images.

So in the picture frame example, you might chop the image into two smaller images and raster the top and left side in one image, and the bottom and right in another. The potential issue is that when you move the head back to the origin point, any sloppiness in the hardware or loss of position might result in the two images not aligning exactly.

This is image analysis, and so would need to be done in Numpy or similar where it is done in compiled code using all cores in parallel.

Another image optimisation (but not needed for rasters converted to images) which would be useful for cut inner first processing, is to remove any vertical and horizontal white margins in the image. This could be done at load time at the same time as changing white to transparent - though the change of dimensions (but not image size) might be confusing if the user has specifically sized the image with margins in e.g. Inkscape.

Reply:
I'm adding but not totally done with `tool imagecut` which will let you manually chop up your images in place. Basically this would let you slice your image with a line. There's a still broken version of this code in main (and mostly hiding as a hidden feature). Also `tool line` but that works fine.

Reply:
I imagine that there may be some overlap of code here.